vdom/core.py

```diff
@@ -72,28 +72,37 @@ class VDOM(object):
     >>> from vdom.helpers import h1
     >>> h1('Hey')
     """
-    # This class should only have these 4 attributes
-    __slots__ = ['tag_name', 'attributes', 'children', 'key', '_frozen']
+    # This class should only have these 5 attributes
+    __slots__ = ['tag_name', 'attributes', 'style', 'children', 'key', '_frozen']
 
-    def __init__(self, tag_name, attributes=None, children=None, key=None, schema=None):
+    def __init__(self, tag_name, attributes=None, style=None, children=None, key=None, schema=None):
         if isinstance(tag_name, dict) or isinstance(tag_name, list):
             # Backwards compatible interface
             warnings.warn('Passing dict to VDOM constructor is deprecated')
             value = tag_name
             vdom_obj = VDOM.from_dict(value)
             tag_name = vdom_obj.tag_name
+            style = vdom_obj.style
             attributes = vdom_obj.attributes
             children = vdom_obj.children
             key = vdom_obj.key
         self.tag_name = tag_name
         self.attributes = FrozenDict(attributes) if attributes else FrozenDict()
         self.children = tuple(children) if children else tuple()
         self.key = key
+        self.style = style if style else {}
 
         # Validate that all children are VDOMs or strings
         if not all([isinstance(c, (VDOM, string_types[:])) for c in self.children]):
             raise ValueError('Children must be a list of VDOM objects or strings')
 
+        # All style keys & values must be strings
+        if not all([
```

Comment:
`[` shouldn't be necessary here (not that it matters much) as all can work against generators